### ISSUE TITLE
Provider produced inconsistent final plan

### ISSUE THREAD

**Issue by jesperharlemnext** [Leaseweb/terraform-provider-leaseweb#13]
When the `leaseweb_dedicated_server` reference is changed and used in the `leaseweb_dedicated_server_installation` it will throw the following error

**Current reference:** srv001.wdc02.lsw.mydomain.com
**Update reference:** srv001.wdc1.lsw.mydomain.com

### Error
```
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for module.wdc1-srv001.leaseweb_dedicated_server_installation.default to include new values learned so far during apply, provider "registry.terraform.io/leaseweb/leaseweb" produced
│ an invalid new value for .hostname: was cty.StringVal("srv001.wdc1.lsw.mydomain.com"), but now cty.StringVal("srv001.wdc02.lsw.mydomain.com").
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```

### Example terraform code
```tf
resource "leaseweb_dedicated_server" "default" {
  reference                       = local.domain_name
  reverse_lookup                  = local.domain_name
  powered_on                      = true
  public_network_interface_opened = true
  public_ip_null_routed           = false
}

resource "leaseweb_dedicated_server_installation" "default" {
  dedicated_server_id = leaseweb_dedicated_server.default.id
  operating_system_id = var.operating_system_id

  hostname = leaseweb_dedicated_server.default.reference
  timezone = "UTC"
  device   = "NVME"

  partition {
    mountpoint = "/boot"
    size       = 1024
    filesystem = "ext2"
  }

  partition {
    mountpoint = "/tmp"
    size       = 8192
    filesystem = "ext4"
  }

  partition {
    size       = 4096
    filesystem = "swap"
  }

  partition {
    mountpoint = "/"
    size       = "*"
    filesystem = "ext4"
  }
}
```

### Workaround
Instead of using `hostname = leaseweb_dedicated_server.default.reference` I'm using `hostname = local.domain_name`

**Comment by nrocco** [comment]
Thanks for reporting this @jesperharlemnext 

We will try to reproduce your issue and discuss how we can get it fixed.

**Comment by mssiebelink** [comment]
Hello @jesperharlemnext 

I like to give an update on this project and the issue in general. 

At the moment we are in the process of moving the terraform plugin to the new Terrafrom plugin framework as well as expanding it. As we speak we are in the process of moving the current implementation for dedicated servers towards the new setup. With this move we will make sure this issue will be resolved.

**Comment by imtiazPabel** [comment]
Hello @jesperharlemnext 

We have released a new tag, `v1.7.0-beta`, which addresses the issues. Thank you for your feedback, and please let us know if you encounter any further problems. 

We're going to close this issue.
